### ISSUE TITLE
Create a dragonfly plugin for MONAILabel

### ISSUE THREAD

**Issue by AlexanderZeilmann** [Project-MONAI/MONAILabel#1637]
[Dragonfly](https://www.theobjects.com/dragonfly/index.html) is a (commercial, but free for academia) software platform for scientific image processing, similar to e.g. 3D Slicer, better/more user friendly in many aspects.

**Describe the solution you'd like**
It would be great to have [dragonfly plugin](https://dev.theobjects.com/dragonfly_2022_2_release/Documentation/Extensions/plugins.html) for MONAI, similar to the [3D Slicer plugin](https://github.com/Project-MONAI/MONAILabel/tree/main/plugins/slicer).

**Describe alternatives you've considered**
An alternative would be to continue working with 3D Slicer just for using MONAI and export/import the data to dragonfly for the many areas where dragonfly is better than 3D Slicer.

**Additional context**
dragonfly offers a similar approach to MONAI in their [dragonfly vizserver](https://www.theobjects.com/dragonfly/dragonfly-vizserver.php).

**Comment by diazandr3s** [comment]
Hi @AlexanderZeilmann,

This sounds interesting! I'm not very familiar with the Draginfly viewer. Do you know if there is documentation on how to modify the UI or add functionalities/modules/plug-ins to it?

In which areas do you think Dragonfly is better than 3DSlicer?

Please let us know

**Comment by lassoan** [comment]
@AlexanderZeilmann The open-source community welcomes companies who would like to commercialize the developed tools, as it is better for everyone if we work together. However, please try to:
- be nice (do not try to promote your software at the expense of some open-source projects, there is room for everyone to be successful)
- be a useful member of the community (show that you are here not to just take but also to give, especially when nobody lnows you yet: do not expect someone to develop features for your commercial software without offering funding, time from your developers, or anything else that is useful for the community; offering your software for free to use to people who would never pay for it anyway is not a community contribution)

**Comment by AlexanderZeilmann** [comment]
> Do you know if there is documentation on how to modify the UI or add functionalities/modules/plug-ins to it?

Yes, for example here https://dev.theobjects.com/dragonfly_2022_2_release/Documentation/Extensions/howtodevelopui.html#howtodevelopui

----
Dear @lassoan,
I have no affiliation with Dragonfly or any other commercial software, and I'm unsure why you might think otherwise. Personally, I find Dragonfly preferable to 3DSlicer due to its more intuitive interface, though this is solely my opinion and not intended as a promotion. I simply believe in the value of having more alternatives available.

I understand that I'm not yet known within the MONAI community, but everyone has to begin somewhere, right? I recognize that developing such a plugin would be a significant undertaking, so I wouldn't embark on it without first discussing it here. This issue isn't about delegating work to others but rather serves as a starting point for discussion.

**Comment by lassoan** [comment]
Your post sounded like Dragonfly marketing, so I assumed you were affiliated with the company. But if you are just an enthusiastic user, then what you wrote was fine. Maybe next time you write about software like this, it is worth adding a sentence to clarify that you are not affiliated.

> I understand that I'm not yet known within the MONAI community.

Sorry, I did not mean you. I meant Dragonfly software and developers are unknown to this community. Therefore, probably at this point, you cannot expect to get much help from developers here. Instead, you can ask Dragonfly if they were interested in developing integration with MONAILabel. If they are open to the idea, then they could assign one of their developers to this task. If they do not have the capacity to work on this or want to keep using their proprietary solutions, then probably you have to work on this yourself (or switch to Slicer and realize just how much better it is than Dragonfly, in large part due to the amazing user and developer community around it).
